From 03e11f0425285bcd14848de2f64fa94bb171504f Mon Sep 17 00:00:00 2001
From: xplodwild <xplod@ouverta.fr>
Date: Thu, 3 Oct 2013 13:08:24 +0200
Subject: [PATCH] Settings: Add lockscreen ring battery setting

Change-Id: Ia07ecc1feedee71b8ea217c53244fdef9ff1bc18

Conflicts:
	res/values/custom_strings.xml
	res/xml/security_settings_slide_delay_cyanogenmod.xml
	src/com/android/settings/SecuritySettings.java
---
 res/values/cm_strings.xml                             |  1 +
 res/xml/security_settings_slide_delay_cyanogenmod.xml |  5 +++++
 src/com/android/settings/SecuritySettings.java        | 15 +++++++++++++++
 3 files changed, 21 insertions(+)

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 2b68a40..9a4a6ec 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -295,6 +295,7 @@
     <string name="unlock_camera_unlock_summary">Pressing the camera button unlocks the device</string>
     <string name="unlock_quick_unlock_control_title">Quick unlock</string>
     <string name="unlock_quick_unlock_control_summary">Unlock automatically when the correct PIN/password is entered</string>
+    <string name="battery_around_lockscreen_ring">Battery level around lock ring</string>
 
     <!-- Slide lock delay -->
     <string name="delay_and_timeout_title">Lock delay</string>
diff --git a/res/xml/security_settings_slide_delay_cyanogenmod.xml b/res/xml/security_settings_slide_delay_cyanogenmod.xml
index e688de5..4fd1004 100644
--- a/res/xml/security_settings_slide_delay_cyanogenmod.xml
+++ b/res/xml/security_settings_slide_delay_cyanogenmod.xml
@@ -41,6 +41,11 @@
             android:entryValues="@array/lockscreen_delay_values"
             android:dependency="screen_lock_slide_delay_toggle" />
 
+        <CheckBoxPreference
+            android:key="battery_around_lockscreen_ring"
+            android:title="@string/battery_around_lockscreen_ring"
+            android:defaultValue="true" />
+
     </PreferenceCategory>
 
 </PreferenceScreen>
diff --git a/src/com/android/settings/SecuritySettings.java b/src/com/android/settings/SecuritySettings.java
index 06829ff..85b13e5 100644
--- a/src/com/android/settings/SecuritySettings.java
+++ b/src/com/android/settings/SecuritySettings.java
@@ -105,6 +105,9 @@ public class SecuritySettings extends SettingsPreferenceFragment
     private static final String KEY_BLACKLIST = "blacklist";
     private static final String CATEGORY_ADDITIONAL = "additional_options";
 
+    // Omni Additions
+    private static final String BATTERY_AROUND_LOCKSCREEN_RING = "battery_around_lockscreen_ring";
+
     private PackageManager mPM;
     DevicePolicyManager mDPM;
 
@@ -137,6 +140,9 @@ public class SecuritySettings extends SettingsPreferenceFragment
     private ListPreference mSmsSecurityCheck;
     private PreferenceScreen mBlacklist;
 
+    // Omni Additions
+    private CheckBoxPreference mLockRingBattery;
+
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
@@ -280,6 +286,12 @@ public class SecuritySettings extends SettingsPreferenceFragment
             CheckBoxPreference vibratePref = (CheckBoxPreference)
                     findPreference(Settings.System.LOCKSCREEN_VIBRATE_ENABLED);
 
+            // Add the additional Omni settings
+            mLockRingBattery = (CheckBoxPreference) root
+                    .findPreference(BATTERY_AROUND_LOCKSCREEN_RING);
+            mLockRingBattery.setChecked(Settings.System.getInt(resolver,
+                    Settings.System.BATTERY_AROUND_LOCKSCREEN_RING, 0) == 1);
+
             final int deviceKeys = res.getInteger(
                     com.android.internal.R.integer.config_deviceHardwareKeys);
             final PreferenceGroup additionalPrefs =
@@ -772,6 +784,9 @@ public class SecuritySettings extends SettingsPreferenceFragment
             lockPatternUtils.setVisibleDotsEnabled(isToggled(preference));
         } else if (KEY_POWER_INSTANTLY_LOCKS.equals(key)) {
             lockPatternUtils.setPowerButtonInstantlyLocks(isToggled(preference));
+        } else if (preference == mLockRingBattery) {
+            Settings.System.putInt(getActivity().getApplicationContext().getContentResolver(),
+                    Settings.System.BATTERY_AROUND_LOCKSCREEN_RING, isToggled(preference) ? 1 : 0);
         } else if (preference == mShowPassword) {
             Settings.System.putInt(getContentResolver(), Settings.System.TEXT_SHOW_PASSWORD,
                     mShowPassword.isChecked() ? 1 : 0);
-- 
1.8.4.2

